### PR TITLE
Feature/redirect root

### PR DESCRIPTION
**Pull request by fceller**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.7, 3.8 (PR not yet available)

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [x] Docs PR: https://github.com/arangodb/docs/pull/696
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket number: ES-882
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [x] Added entry to *Release Notes* 
- [x] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by dothebart** [comment]
while its not whats demanded here: https://github.com/arangodb/arangodb/issues/8176 it could solve the initial issue the user has here.

File: arangod/GeneralServer/GeneralServerFeature.cpp [hunk 1]
```diff
@@ -299,6 +309,14 @@ Result GeneralServerFeature::reloadTLS() {  // reload TLS data from disk
   return res;
 }
```
Comment (dothebart) [inline arangod/GeneralServer/GeneralServerFeature.cpp:311]: don't we do one line getters in the .h?

**Review by dothebart — APPROVED** [review]
maybe pls. move the implementation to the header. 
This adds 4 single server starts - plus most probably also 4 cluster starts? 
Couldn't at least some of the tests be integrated with others already existing server starts?

**Review by jsteemann — APPROVED** [review]
LGTM!

File: arangod/GeneralServer/GeneralServerFeature.cpp [hunk 2]
```diff
@@ -175,6 +177,14 @@ void GeneralServerFeature::collectOptions(std::shared_ptr<ProgramOptions> option
                      "trusted origin URLs for CORS requests with credentials",
                      new VectorParameter<StringParameter>(&_accessControlAllowOrigins));
 
+  options->addOption("--http.redirect-root-to",
+                    "redirect of root URL",
+                    new StringParameter(&_redirectRootTo));
```
Comment (jsteemann) [inline arangod/GeneralServer/GeneralServerFeature.cpp:182]: This new option, as well as the following one, should get a `.setIntroducedIn(30712)` extension.

**Pull request merged**